### PR TITLE
Created POST route for adding admins

### PR DESCRIPTION
**Pull request by N00Bmaster2810**
## Issue that this pull request solves

 Closes: #79 under SWOC

###

- Add a route for creating admins.[POST]
- Validation Schema for the request body
- API Testing using Postman demonstrating all possible cases(valid, invalid) with test cases

## Types of changes

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] My changes does not break the current system and it passes all the current test cases.

## Screenshots
Screenshot (275)
Screenshot (276)
Screenshot (277)

## Other information

Any other information that is important to this pull request

### CODE REVIEW

**Comment by jackfrost13** [comment]
@N00Bmaster2810 You need to share postman json file. Please update that. In the meanwhile, I will be reviewing your code

File: backend/index.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const helmet = require('helmet');
 const http = require('http');
 const bodyParser = require('body-parser');
 const responseTime = require('response-time');
-const routes = require('./app/routes');
+const routes = require('./app/routes/');
```
Comment (jackfrost13) [inline backend/index.js:7]: Please revert this to original state so that it is free from linter errors

File: backend/app/routes/index.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ const router = require('express').Router({ mergeParams: true });
 const admin = require('./admin');
 const auth = require('./auth');
 
-router.use('/admin', admin);
-router.use('/auth', auth);
+router.use(admin);
```
Comment (jackfrost13) [inline backend/app/routes/index.js:5]: Revert this to the original state. This will break our system

File: backend/app/routes/index.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ const router = require('express').Router({ mergeParams: true });
...
+router.use(auth);
```
Comment (jackfrost13) [inline backend/app/routes/index.js:6]: Revert this to the original state. This will break our system

File: backend/app/routes/admin/index.js [hunk 4]
```diff
@@ -3,12 +3,18 @@ const validationMiddleware = require('../../../helpers/middlewares/validation');
...
+const postAdmin = require('./postAdmin');
 
 router.get('/createSuperAdmin', createSuperAdmin);
 router.post(
   '/superAdmin',
   validationMiddleware(postSuperAdminSchema),
   postSuperAdmin
 );
+router.post(
+  '/admin',
```
Comment (jackfrost13) [inline backend/app/routes/admin/index.js:15]: here the path should be just `/`

File: backend/app/routes/admin/index.js [hunk 5]
```diff
@@ -3,12 +3,18 @@ const validationMiddleware = require('../../../helpers/middlewares/validation');
...
+  validationMiddleware(postSuperAdminSchema),
```
Comment (jackfrost13) [inline backend/app/routes/admin/index.js:16]: This will be an authorised API. Only existing superAdmins can add new admins

**Review by jackfrost13 — CHANGES_REQUESTED** [review]

**Comment by jackfrost13** [comment]
@N00Bmaster2810 Please go through my comments and add a comprehensive API testing file

File: backend/app/routes/admin/index.js (hunk 5 above)
Comment (N00Bmaster2810) [reply backend/app/routes/admin/index.js:16]: Sir so please can you suggest what should I do , remove the use of this mentioned middleware and make another one or what?

Comment (jackfrost13) [reply backend/app/routes/admin/index.js:16]: Go through the auth module. You need to use the auth middleware also. Moreover, during the post request, you need to check if the request is coming from a superAdmin or not

Comment (N00Bmaster2810) [reply backend/app/routes/admin/index.js:16]: ok sir

**Comment by jackfrost13** [comment]
@N00Bmaster2810 Please run the linter before committing and add API Testing files, only then I can help with you a review

**Comment by jackfrost13** [comment]
@N00Bmaster2810 Is the API Testing file present?

File: backend/app/routes/admin/index.js [hunk 6]
```diff
@@ -21,5 +22,11 @@ router.post(
   validationMiddleware(postSuperAdminSchema),
   postSuperAdmin
 );
+router.post(
+  '/',
+  validationMiddleware(postSuperAdminSchema),
+  authMiddleware,
+  postAdmin
```
Comment (jackfrost13) [inline backend/app/routes/admin/index.js:29]: This throws an error. This is not defined. Please import the file.

**Review by jackfrost13 — CHANGES_REQUESTED** [review]
Please add API Testing file without that I cannot move ahead with the review

**Comment by N00Bmaster2810** [comment]
> Please add API Testing file without that I cannot move ahead with the review

Sir, I was learning about hot to do this API testing file thing but couldn't understant it could you please help regarding this.

**Comment by N00Bmaster2810** [comment]
> Please add API Testing file without that I cannot move ahead with the review

Sir, I was learning about how to do this API testing file thing but couldn't understant it could you please help regarding this.

**Comment by jackfrost13** [comment]
> > Please add API Testing file without that I cannot move ahead with the review
> 
> Sir, I was learning about how to do this API testing file thing but couldn't understant it could you please help regarding this.

Sure, just check the resources provided in the API testing issue reference in #79. Let me know If you need more help. Feel free to shoot your doubts here.

**Comment by N00Bmaster2810** [comment]
@jackfrost13 I have made the changes
https://www.getpostman.com/collections/aba0b43b7e2e18f7d9a7
This is the link to the test collection

**Review by jackfrost13 — APPROVED** [review]

**Comment by jackfrost13** [comment]
@N00Bmaster2810 I have made a few changes, go through it.
I am merging this now

**Pull request merged**